components/ui/Feed/FeedView.tsx

```diff
@@ -80,156 +54,96 @@ function FeedView({ feed, community = false, header }: FeedViewProps) {
     []
   );
 
-  useEffect(() => {
-    navigation.setOptions({
-      headerRight: () => headerRight(),
-    });
-  }, [feed.sort]);
-
-  useEffect(() => {
-    if (!feed.posts || feed.posts.length < 1 || communityId.current !== 0)
-      return;
-    communityId.current = feed.posts[0].community.id;
-    communityName.current = feed.posts[0].community.name;
-  }, [feed.posts]);
+  const firstPost = (feed.posts?.length ?? 0) > 0 ? feed.posts[0] : undefined;
 
-  const onSortPress = () => {
-    const options = [
-      "Top Day",
-      "Top Week",
-      "Hot",
-      "Active",
-      "New",
-      "Most Comments",
-      "Cancel",
-    ];
-    const cancelButtonIndex = 6;
+  const postCommunity: Community = useMemo(() => {
+    if (!firstPost || !community) return undefined;
```

Comment:
instead of using a side effect to update 2 refs for maintaining the community ID and community Name, we can derive this information from the first post and the `community` flag being passed in. Doing this allows us to update the header as part of a normal render pass, if necessary.